pyomo/contrib/mindtpy/nlp_solve.py

```diff
@@ -185,6 +185,11 @@ def handle_subproblem_optimal(fixed_nlp, solve_data, config, cb_opt=None):
         for c in fixed_nlp.tmp_duals:
             if fixed_nlp.dual.get(c, None) is None:
                 fixed_nlp.dual[c] = fixed_nlp.tmp_duals[c]
+            elif (
+                config.nlp_solver == 'cyipopt'
+                and solve_data.objective_sense == minimize
+            ):
+                fixed_nlp.dual[c] = -fixed_nlp.dual[c]
```

Comment:
Duals workaround. Maybe just add a comment in all these places citing the issue # for future reference